File: 240305/k번째로 신기한 문자열/kth-special-string.py
n, k, T = input().split()
n = int(n)
k = int(k)

word_list = []

for i in range(n):
  word = input()
  if T in word:
    word_list.append(word)

word_list.sort()
print(word_list[k-1])

File: 240421/k번째로 신기한 문자열/kth-special-string.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split(`\n`)

const first = input[0].split(' ')

const n = Number(first[0])
const k = Number(first[1])
const T = first[2]

let TList = []

for (let i = 1; i < n; i++ ) {
    if (T === input[i].slice(0,2)) {
        TList.push(input[i])
    }
}

TList.sort()

console.log(TList[k-1])